### DISCUSSION TITLE
Downloading Landsat - Any workaround? Other sources?

### DISCUSSION THREAD

**Discussion by naq979** [semiautomaticgit/SemiAutomaticClassificationPlugin#288] [Q&A] [answered]
I'm aware that the current version has issues with downloading Landsat imagery but would it be possible to work around that and download the imagery from other sources? For example, if I were to first create a cloud-masked composite on GEE and export it to a directory as multi-band rasters, would that work? What kind of file requirements would be needed to use it in the plug-in? How about metadata?

**Comment by semiautomaticgit** [comment, marked as answer]
Hello @naq979 , basically you can use any raster in SCP. Classifications work better with single band rasters.
You can download Landsat images from the USGS website or from GEE.
If you need to preprocess the images in SCP, also the metadata are required (in the format provided in Landsat Collection 2).

**Reply by naq979** [reply]
Thanks @semiautomaticgit ! I've tried it out and it works :) I've done one round of isodata clustering with 10 classes, however, when using the same bandset and changing the number of classes, I faced this error: AttributeError: 'NoneType' object has no attribute 'GetRasterBand'. The error happens again even when i try a different bandset. Does the problem lie with the single band rasters I used as input? Or is it a system error?

[error.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/12776828/error.txt)
[log.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/12776829/log.txt)

**Reply by semiautomaticgit** [reply]
You are welcome @naq979 .

About the issue, unfortunately the log file is empty. 
However, you could try to split the bands and create a new Band set, then try again.
